diff --git a/fastboot/instance-initializers/ember-data-fastboot.js b/fastboot/instance-initializers/ember-data-fastboot.js
index 3aa89511a5e4c24bc94d9705895be8c689ea5748..b6a1445b5ee83b6aa0c068905c45dbe9d15da9a1 100644
--- a/fastboot/instance-initializers/ember-data-fastboot.js
+++ b/fastboot/instance-initializers/ember-data-fastboot.js
@@ -1,15 +1,39 @@
+import Ember from 'ember';
+
 export function initialize(applicationInstance) {
   let store = applicationInstance.lookup('service:store');
   let shoebox = applicationInstance.lookup('service:fastboot').get('shoebox');
-  const modelNames = applicationInstance.lookup('data-adapter:main').getModelTypes().mapBy('name');
+
+  // Workaround for dataAdapter.getModelTypes() no longer working
+  const dataAdapter = applicationInstance.lookup('data-adapter:main');
+  const debugAdapter = dataAdapter.get('containerDebugAdapter');
+  const modelNames = debugAdapter.catalogEntriesByType('model');
 
   shoebox.put('ember-data-store', {
     get records() {
       return modelNames.map(name => {
-        return store.peekAll(name).toArray();
+        return store.peekAll(name);
       }).reduce((a,b) => a.concat(b), [])
         .filter(record => record.get('isLoaded') && !record.get('isNew'))
-        .map(record => record.serialize({ includeId: true}))
+        .map(record => {
+          const serializedRecord = record.serialize({ includeId: true});
+
+          record.eachRelationship((name, meta) => {
+            const link = record[meta.kind](name).link();
+
+            if (link) {
+              const dashName = Ember.String.dasherize(name);
+
+              serializedRecord.data.relationships = serializedRecord.data.relationships || {}
+              serializedRecord.data.relationships[dashName] = serializedRecord.data.relationships[dashName] || {}
+              serializedRecord.data.relationships[dashName].links = {
+                related: link
+              }
+            }
+          })
+
+          return serializedRecord
+        })
         .reduce((a,b) => { a.data.push(b.data); return a; }, { data: [] });
     }
   });
diff --git a/yarn.lock b/yarn.lock
deleted file mode 100644
index 3a34216b04a8bbd5d4ba576a97ecf27c498ca7d1..0000000000000000000000000000000000000000
